test/library/packages/UnitTest/Launcher.chpl

```diff
@@ -107,12 +141,17 @@ module Launcher {
         if temp[1].strip().endsWith("]") {
           var strSplit = temp[1].strip().split("]");
           var testNameIndex = strSplit[1].split("[");
-          testName = fileName+": "+testNameIndex[1];
+          testName = testNameIndex[1];
           curIndex = testNameIndex[2]: int;
           result.startTest();
           if temp.size > 1 {
             flavour = temp[2].strip();
-            if flavour == "OK" then result.addSuccess(testName);
+            if flavour == "OK" {
+              result.addSuccess(fileName+": "+testName);
```

Comment:
> You do this same operation in at least four different places. Should it be a conversion that addSuccess/addFailure/etc perform on their own, maybe by taking the fileName as an additional argument?

I didn't get what you meant.

Although now I am doing it in one function.

Reply:
It looks like you have adjusted for performing the `fileName+": "+testName` operation multiple times, which was what I was referring to.

My point was more that if it's something you want consistently done for `addSuccess`, `addFailure`, maybe the operation should be performed inside those functions, by taking the file name and the test name as arguments?  Otherwise you could forget to do it for some calls, and end up with some printing `test.chpl: foo` while others only print `foo`?  It's easier to forget to change the argument from `foo` to `test.chpl: foo` than it is to forget to pass two arguments (since the compiler will yell at you for passing the wrong number of arguments)

Reply:
Okay Ma'am.